pkg/querier/queryrange/codec.go

```diff
@@ -744,23 +741,104 @@ func (Codec) MergeResponse(responses ...queryrangebase.Response) (queryrangebase
 		}, nil
 	case *LabelVolumeResponse:
 		resp0 := responses[0].(*LabelVolumeResponse)
-		headers := resp0.Headers
+		headers := make([]*definitions.PrometheusResponseHeader, len(resp0.Headers))
+		for i, header := range resp0.Headers {
+			h := header
+			headers[i] = &h
+		}
 
 		resps := make([]*logproto.LabelVolumeResponse, 0, len(responses))
 		for _, r := range responses {
 			resps = append(resps, r.(*LabelVolumeResponse).Response)
 		}
 
-		return &LabelVolumeResponse{
-			Response: labelvolume.Merge(resps, resp0.Response.Limit),
-			Headers:  headers,
-		}, nil
+		promResponse := queryrangebase.PrometheusResponse{
+			Status:  loghttp.QueryStatusSuccess,
+			Data:    MergeToPrometheusResponse(resps, resp0.Response.Limit),
+			Headers: headers,
+		}
 
+		return &LokiPromResponse{
+			Response:   &promResponse,
+			Statistics: stats.Result{},
+		}, nil
 	default:
 		return nil, errors.New("unknown response in merging responses")
 	}
 }
 
+func MergeToPrometheusResponse(responses []*logproto.LabelVolumeResponse, limit int32) queryrangebase.PrometheusData {
+	mergedVolumes := labelvolume.MergeVolumes(responses...)
```

Comment:
I think we could turn this into a sorted list of `Volume` to simplify the next step